cdap-ui/app/cdap/components/PipelineConfigurations/Store/index.js

```diff
@@ -197,8 +197,8 @@ const getRuntimeArgsForDisplay = (currentRuntimeArgs, macrosMap) => {
     currentRuntimeArgs.pairs.forEach((currentPair) => {
       let key = currentPair.key;
       runtimeArgsMap[key] = currentPair.value || '';
-      if (currentPair.notDeletable && currentPair.provided) {
-        providedMacros[key] = currentPair.value;
+      if (currentPair.uniqueId && Object.prototype.hasOwnProperty.call(macrosMap, key)) {
+        macrosWithIds[key] = currentPair.uniqueId;
```

Comment:
Can't we combine this and `const macros = ` statement into one? Do we have to reset `currentRuntimeArgs.pairs` 3 times?

Reply:
Simplified the implementation here. Did not change the implementation in hydrator-service as that needs more work and reusing this implementation won't work there.